src/bp/core/services/bot-monitoring-service.ts

```diff
@@ -0,0 +1,116 @@
+import 'bluebird-global'
+import { Logger, LoggerEntry } from 'botpress/sdk'
+import { ConfigProvider } from 'core/config/config-loader'
+import { LogsRepository } from 'core/repositories/logs'
+import { TYPES } from 'core/types'
+import { inject, injectable, tagged } from 'inversify'
+import { Redis } from 'ioredis'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import Redlock from 'redlock'
+
+import { JobService } from './job-service'
+
+@injectable()
+export class BotMonitoringService {
+  private _redisClient?: Redis
+  private _redlock!: Redlock
+  private _keyMonitorLock = 'bot_monitor_lock'
+  private _lock
+
+  private _intervalRef: NodeJS.Timeout | undefined
+  private _currentPromise
+  private _lastBotMonitorCheckTs?: Date
+  private _interval?: number
+
+  public onBotError?: (botId: string, events: LoggerEntry[]) => Promise<void>
+
+  constructor(
+    @inject(TYPES.Logger)
+    @tagged('name', 'BotMonitoring')
+    protected logger: Logger,
+    @inject(TYPES.ConfigProvider) private configProvider: ConfigProvider,
+    @inject(TYPES.LogsRepository) private logsRepo: LogsRepository,
+    @inject(TYPES.JobService) private jobService: JobService
+  ) {}
+
+  async start() {
+    const config = await this.configProvider.getBotpressConfig()
+    if (!config?.botMonitoring?.enabled) {
+      return
+    }
+
+    this._redisClient = this.jobService.getRedisClient()
+
+    if (this._redisClient) {
+      this._redlock = new Redlock([this._redisClient], {
```

Comment:
since most of those are default params can't you simply pass the retryCount instead passing the whole config object ?